crates/ursa-store/src/store.rs

```diff
@@ -169,6 +178,32 @@ where
         }
         Ok(res)
     }
+
+    /// Build a temporary car file from a root cid and return the size
+    async fn car_size(&self, root_cid: &Cid) -> Result<u64> {
```

Comment:
This behavior is expensive, have we benchmarked this?

Reply:
It could probably be improved by manually measuring a car file and creating a formula to directly calculate the size from the content lengths instead of constructing a full car

Reply:
I think we should traverse the blockstore and get the blob size not create a new file

Reply:
Don't we need the car file size not blob size for gateway tho?

Reply:
made some improvements to the method
benchmarked 1, 10, 100 MB car files at 1ms, 12ms, and 150ms respectively with the optimizations